inference-engine/src/transformations/src/transformations/op_conversions/einsum_decomposition.cpp

```diff
@@ -0,0 +1,643 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/einsum_decomposition.hpp"
+
+#include <memory>
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "itt.hpp"
+
+namespace {
+/// \brief      Check if the transformation is applicable to Einsum with a given equation
+///
+/// \param      subscript          A subscript to check its format
+///
+/// \return     true - applicable, false - not applicable
+///
+bool is_subscript_applicable(const std::string& subscript) {
+    auto labels = ngraph::op::v7::Einsum::extract_labels(subscript);
+    std::unordered_set<std::string> met_labels;
+    for (auto const& label : labels) {
+        if (label == "..." || met_labels.find(label) != met_labels.end()) {
+            return false;
+        }
+        met_labels.insert(label);
+    }
+    return true;
+}
+
+/// \brief      Compute einsum_path for a given Einsum node meaning that the (pseudo-)optimal
+/// order of operands contraction in terms of performance and memory consumption
+///
+/// \param      einsum_node         An input Einsum node
+///
+/// \return     a vector of pairs with input indices assuming that the intermediate result is
+/// appended in the tail
+///
+std::vector<std::pair<size_t, size_t>> compute_einsum_path(std::shared_ptr<const ngraph::opset7::Einsum> einsum_node) {
+    // TODO: implement greedy algorithm for finding pseudo-optimal einsum_path
+    std::vector<std::pair<size_t, size_t>> einsum_path;
+    auto num_inputs = einsum_node->get_input_size();
```

Comment:
```suggestion
    const int64_t num_inputs = einsum_node->get_input_size();
```
Just not to overflow on  `num_inputs - 1`

Reply:
Added assert